spacialgaze-plugins/exp.js

```diff
@@ -0,0 +1,284 @@
+'use strict';
+
+/********************************************
+*     EXP SYSTEM FOR POKEMON SHOWDOWN	    *
+*     	  By Volco and HoeenHero	    *
+********************************************/
+
+const DEFAULT_AMOUNT = 0;
+
+function isExp(exp) {
+	let numExp = Number(exp);
+	if (isNaN(exp)) return "Must be a number.";
+	if (String(exp).includes('.')) return "Cannot contain a decimal.";
+	if (numExp < 1) return "Cannot be less than one EXP.";
+	return numExp;
+}
+SG.isExp = isExp;
+
+let EXP = SG.EXP = {
+	readExp: function (userid, callback) {
+		if (typeof callback !== 'function') {
+			throw new Error("EXP.readExp: Expected callback parameter to be a function, instead received " + typeof callback);
+		}
+
+		// In case someone forgot to turn `userid` into an actual ID...
+		userid = toId(userid);
+
+		let amount = Db.exp.get(userid, DEFAULT_AMOUNT);
+		return callback(amount);
+	},
+
+	writeExp: function (userid, amount, callback) {
+		// In case someone forgot to turn `userid` into an actual ID...
+		userid = toId(userid);
+
+		// In case someone forgot to make sure `amount` was a Number...
+		amount = Number(amount);
+		if (isNaN(amount)) {
+			throw new Error("EXP.writeExp: Expected amount parameter to be a Number, instead received " + typeof amount);
+		}
+		let curTotal = Db.exp.get(userid, DEFAULT_AMOUNT);
+		Db.exp.set(userid, curTotal + amount);
+		let newTotal = Db.exp.get(userid);
+		if (callback && typeof callback === 'function') {
+			// If a callback is specified, return `newTotal` through the callback.
+			return callback(newTotal);
+		}
+	},
+};
+
+function addExp(user, room, amount) {
+	user = Users.getExact(toId(user));
+	EXP.readExp(user.userid, totalExp => {
+		let oldLevel = SG.level(user);
+		EXP.writeExp(user.userid, amount);
+		if (!user || !room) return;
+		user.sendTo(room, 'You have gained ' + amount + ' EXP.');
+		let level = SG.level(user);
+		if (oldLevel < level) {
+			//let rewardLevel;
+			let currency;
+			if (level === 3) {
+				//rewardLevel = '';
+				currency = 2;
+				Economy.writeMoney(user.userid, currency, () => {
+					Economy.readMoney(user.userid, newAmount => {
+						Economy.logTransaction(Chat.escapeHTML(user.userid) + ' has received ' + currency + ' ' + (currency === 1 ? global.currencyName : global.currencyPlural) + ' from a level up.');
+					});
+				});
+				//user.sendTo(room, 'You have earned ' + rewardLevel + ' for level up!');
+			} else if (level === 5) {
+				//rewardLevel = '';
+				currency = 5;
+				Economy.writeMoney(user.userid, currency, () => {
+					Economy.readMoney(user.userid, newAmount => {
+						Economy.logTransaction(Chat.escapeHTML(user.userid) + ' has received ' + currency + ' ' + (currency === 1 ? global.currencyName : global.currencyPlural) + ' from a level up.');
+					});
+				});
```

Comment:
If you want to see the new amount, you just need to use `Economy.writeMoney` and pass `newAmount` through the callback.